### ISSUE TITLE
Logging in without github

### ISSUE THREAD

**Issue by Daggerbot** [rust-lang/crates.io#1411]
Is it or will it be possible to log in to crates.io without a github account? I'll be deleting my account in the near future and I'm currently moving my projects over to GitLab. If this is not possible, I'll have to hand over my biggest project to a new maintainer ([x11-rs](https://github.com/daggerbot/x11-rs)), but I'm not sure how to even find somebody to take it over.

**Comment by steveklabnik** [comment]
duplicate of https://github.com/rust-lang/crates.io/issues/326, thanks!

**Comment by 89z** [comment]
Can you please unlock the other issue? If you're gonna point people there, wouldn't it make sense to point people to somewhere they can actually talk about it?

**Comment by 89z** [comment]
CC @alexcrichton

**Comment by 89z** [comment]
CC @jtgeibel

**Comment by 89z** [comment]
Can you please unlock the other issue? If you're gonna point people there, wouldn't it make sense to point people to somewhere they can actually talk about it? CC:

@alexcrichton
@carols10cents
@jtgeibel

**Comment by Turbo87** [comment]
feel free to open a [discussion](https://github.com/rust-lang/crates.io/discussions/new) for this topic :)

**Comment by 89z** [comment]
Its not really a discussion, its an issue. Currently only GitHub logins are supported. Unless crates.io is a GitHub property (its not), then having only one login method is dubious. Previous reasons were more or less that "no one had worked on it". Fair enough, but that was 6 years ago, and the issue while still open, is locked. At any rate, here is my quarter into the wishing well:

https://github.com/rust-lang/crates.io/discussions/5253

**Comment by Turbo87** [comment]
> I think its more an issue than discussion

maybe, but it's not directly actionable. allowing other forms of authentication has quite a few side-effects that will need to be figured out before we can do any implementation work. 

e.g. how do we prevent spam accounts? how do we deal with username conflicts? how do we migrate the GitHub-based teams into crates.io teams and keep them in sync?

it's unfortunately not quite as trivial as just adding another OAuth provider 🤷‍♂️

**Comment by 89z** [comment]
> maybe, but it's not directly actionable

why do you approach this situation, like Rust is the only programming language thats ever existed? Have you done any research into other platforms, to see how they solved your concerns?

**Comment by Turbo87** [comment]
I believe there has been some research done a while ago by other team members, but I personally haven't looked at it in detail. As I mentioned, we have a few other items on the todo list that have higher priority for now.

**Comment by 89z** [comment]
So, why do you feel you are in a position to make this comment:

> maybe, but it's not directly actionable

when, by your own admission, you haven't even researched the problem space?
